Delete pods on the node that erorred or timed out on drain

george-angel opened the pull request:
This will still be more elegant then shutting down the node and sleeping
for a static amount of time. Also this change should result in faster
run overall.

Tested in development cluster

george-angel committed:
Delete pods on the node that erorred or timed out on drain

This will still be more elegant then shutting down the node and sleeping
for a static amount of time. Also this change should result in faster
run overall.

george-angel committed:
fmt

george-angel committed:
Wait for ASG to remove nodes that have been terminated

If you immediately resume auto-scaling actions after terminating
instances the ASG is still under impression it has 2x nodes (half of
them unhealthy), several processes will start at the same time.

Namely "ReplaceUnhealthy" will start spinning up new nodes, and when the
"ScheduledActions" catches up and starts scaling in, your now "old"
nodes will be terminated.

ffilippopoulos commented on kube-aws-updater:
```diff
@@ -129,77 +130,118 @@ label_for_cycling() {
+    sleep 16
+    aws ${aws_opts} --output=json autoscaling describe-auto-scaling-groups \
+      --auto-scaling-group-names "${asg_name}" |\
+      jq -e -r '."AutoScalingGroups"[0]."Instances"[] | select (."LifecycleState" == "Terminating")'
```
I am thinking if that would better be something like:
```
aws ${aws_opts} --output=json autoscaling describe-auto-scaling-groups --auto-scaling-group-names "${asg_name}" | jq -e '."AutoScalingGroups"[0]."Instances" | length'
```
and then compare it with `${asg_count}` (maybe in the while loop if that oneliner doesn't look too bad). 
I am just a bit concerned if exit code 4 is something meaningful here.
```
-e / --exit-status:

Sets the exit status of jq to 0 if the last output values was neither false nor null, 1 if the last output value was either false or null, or 4 if no valid result was ever produced
```
from https://stedolan.github.io/jq/manual/

ffilippopoulos commented on kube-aws-updater:
```diff
@@ -129,77 +130,118 @@ label_for_cycling() {
 
   # - stop ASG actions
-  aws ${aws_opts} --output=json autoscaling suspend-processes --auto-scaling-group-name "${asg_name}"
+  aws ${aws_opts} --output=json autoscaling suspend-processes \
```
Maybe a comment that we actually need to keep `Terminate` enabled here would be handy for people that are not aware how we decide what to suspend. As the first thing that comes in mind is to disable all.

george-angel committed:
PR feedback

 - Wait for total number of instances instead of counting Terminated
   instances
 - Explain the selective suspension of auto scaling processes
